engine/src/widget-popup.cpp

```diff
@@ -426,9 +423,12 @@ extern "C" MC_DLLEXPORT_DEF MCValueRef MCWidgetExecPopupAtLocationWithProperties
 	MCGPoint t_point;
 	MCCanvasPointGetMCGPoint(p_at, t_point);
 	
-	MCPoint t_at;
-	t_at = MCGPointToMCPoint(MCWidgetMapPointToGlobal(MCcurrentwidget, t_point));
-	
+    MCWidget *t_host;
+    t_host = MCWidgetGetHost(MCcurrentwidget);
+    
+    MCPoint t_at;
+	t_at = t_host->getstack()->globaltostackloc(MCGPointToMCPoint(MCWidgetMapPointToGlobal(MCcurrentwidget, t_point)));
```

Comment:
@runrevmark I'm wondering if we might need to ensure `t_host->getstack()` is open and visible here or if there might be valid reasons for `popup widget` to be called in LCB when the host is not on screen.

Reply:
Hmm - another issue here is that we are assuming there is a 'current widget' - which precludes an LCB library from using popup widget :|

Reply:
Ah - sorry I just noticed that the whole call is guarded by EnsureCurrentWidget :)

Okay so I think we do need to ensure that the widget has an open and visible host - otherwise the 'at' relative position makes no sense.

Reply:
Hmm... the relative position still makes sense if the widget is not visible doesn't it?

Reply:
As mentioned offline, I meant to say 'open and visible host stack' :)